README.md

```diff
@@ -1,101 +1,101 @@
 
 ```ruby
 knife[:aws_access_key_id] = "Your AWS Access Key ID"
 knife[:aws_secret_access_key] = "Your AWS Secret Access Key"
 ```
 
-If your `knife.rb` file will be checked into a SCM system (ie readable by others) you may want to read the values from environment variables:
+Additionally if using AWS STS:
+
+```ruby
+knife[:aws_session_token] = "Your AWS Session Token"
+```
+
+Note: If your `knife.rb` file will be checked into a source control management system, or is otherwise accessible by others, you may want to use one of the other configuration methods to avoid exposing your credentials.
+
+### Environmental Variables
+Knife-ec2 can also read your credentials from shell environmental variables. Export `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN` variables in your shell then add the following configuration to your `knife.rb` file:
 
 ```ruby
 knife[:aws_access_key_id] = ENV['AWS_ACCESS_KEY_ID']
 knife[:aws_secret_access_key] = ENV['AWS_SECRET_ACCESS_KEY']
-# Optional if you're using Amazon's STS
-knife[:aws_session_token] = ENV['AWS_SESSION_TOKEN']
 ```
 
-You also have the option of passing your AWS API Key/Secret into the individual knife subcommands using the `-A` (or `--aws-access-key-id`) `-K` (or `--aws-secret-access-key`) command options
+Additionally if using AWS STS:
 
-```bash
-# provision a new m1.small Ubuntu 10.04 webserver
-$ knife ec2 server create -r 'role[webserver]' -I ami-7000f019 -f m1.small -A 'Your AWS Access Key ID' -K "Your AWS Secret Access Key"
+```ruby
+knife[:aws_session_token] = ENV['AWS_SESSION_TOKEN']
 ```
 
-If you are working with Amazon's command line tools, there is a good chance
-you already have a file with these keys somewhere in this format:
+### CLI Arguments
+You also have the option of passing your AWS API Key/Secret into the individual knife subcommands using the `-A` (or `--aws-access-key-id`) `-K` (or `--aws-secret-access-key`) command options
 
-    AWSAccessKeyId=Your AWS Access Key ID
-    AWSSecretKey=Your AWS Secret Access Key
+Example of provisioning a new t2.micro Ubuntu 14.04 webserver:
 
+```bash
+$ knife ec2 server create -r 'role[webserver]' -I ami-cd0fd6be -f t2.micro -A 'Your AWS Access Key ID' -K "Your AWS Secret Access Key"
+```
 
-The new config file format used by Amazon's command line tools is also supported:
+### AWS Credential File
+Amazon's newer credential config file format is also supported by knife:
 
-    [default]
-    aws_access_key_id = Your AWS Access Key ID
-    aws_secret_access_key = Your AWS Secret Access Key
+```
+[default]
+aws_access_key_id = Your AWS Access Key ID
+aws_secret_access_key = Your AWS Secret Access Key
+```
 
-In this case, you can point the <tt>aws_credential_file</tt> option to
-this file in your <tt>knife.rb</tt> file, like so:
+In this case, you can point the `aws_credential_file` option to this file in your `knife.rb` file, like so:
 
 ```ruby
 knife[:aws_credential_file] = "/path/to/credentials/file/in/above/format"
```

Comment:
This is probably a more useful example as it is what most people will use:

knife[:aws_credential_file]    = File.join(ENV['HOME'], "/.aws/credentials")